assemblies/assembly_architect-decision-services/states-con.adoc

```diff
@@ -0,0 +1,345 @@
+
+As this is a simple validation use case we will add a single rule to disqualify any applicant younger than 18.
+
+[source]
+----
+package com.company.license
+
+rule "Is of valid age"
+when
+    $a : Applicant( age < 18 )
+then
+    $a.setValid( false );
+end
+----
+
+To make the engine aware of data that will be processed against the rules, we have to insert the data, much like with a database. When the Applicant instance is inserted into the engine it is evaluated against the constraints of the rules. In this case only two constraints for one rule.
+We use two because the `Applicant` type is the first object type constraint and `age < 18` is the second field constraint. An object type constraint plus its zero or more field constraints is referred to as a pattern. When an inserted instance satisfies both the object type constraint and all the field constraints, it is said to be matched.
+
+The `$a` is a binding variable which enables us to reference the matched object in the consequence. The dollar character `$` is optional, but it helps to differentiate variable names from field names. The process of matching patterns against the inserted data is often referred to as `pattern matching`.
+
+To use this rule you must place it in a Drools file as a plain text file with .drl extension. For this example, we will name this file `licenseApplication.drl` and store it in a Kie project. A Kie project has the structure of a normal Maven project with an additional kmodule.xml file that defines the ``KieBase``s and ``KieSession``s that can be created.
+The file must be located in the resources/META-INF folder of the Maven project, while all the other Drools artifacts, such as the `licenseApplication.drl` containing the former rule, must be stored in the resources folder or in any other subfolder.
+
+Since meaningful defaults are provided for all configuration aspects, the simplest `kmodule.xml` file can contain just an empty kmodule tag as shown here:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<kmodule xmlns="http://www.drools.org/xsd/kmodule"/>
+----
+
+At this point it is possible to create a `KieContainer` that reads the files to be built from the classpath.
+
+[source,java]
+----
+KieServices kieServices = KieServices.Factory.get();
+KieContainer kContainer = kieServices.getKieClasspathContainer();
+----
+
+The above code snippet compiles all the DRL files found on the classpath and adds the result of this compilation, a ``KieModule``, in the ``KieContainer``. If there are no errors, we create our session from the `KieContainer` and execute against some data:
+
+[source,java]
+----
+StatelessKieSession kSession = kContainer.newStatelessKieSession();
+Applicant applicant = new Applicant( "Mr John Smith", 16 );
+assertTrue( applicant.isValid() );
+ksession.execute( applicant );
+assertFalse( applicant.isValid() );
```

Comment:
Could we please update do differentiate more rule snippets and java snippets:
```java
Applicant applicant = new Applicant("Mr John Smith", 16);
assertTrue(applicant.isValid());
ksession.execute(applicant);
assertFalse(applicant.isValid());
```

Reply:
@jomarko, Are you able to provide examples? If not, I will see if Edson can.

Reply:
@michelehaglund I think no special example needed. I Just think that java code snippets shouldn't use `whitespace` around braces. So for example in java snippets is better to use `new Applicant("Mr John Smith", 16);` than `new Applicant( "Mr John Smith", 16 );` . Do you see my point please? I know it is just minor difference but I think helps with consistency. 

Also I think whitespace around braces for rule snippets is fine.